### ISSUE TITLE
Tensor Nan Error

### ISSUE THREAD

**Issue by frankchen121212** [scverse/scvi-tutorials#65]
Hello, thank you for the great work, I'm trying to run DestVI on other datasets, and I've transformed all the data into h5ad format with all necessary values to train the model,

I'm following [DestVI_tutorial.ipynb](https://github.com/scverse/scvi-tutorials/blob/master/DestVI_tutorial.ipynb) to run the training in order to get the cell type composition matrix

My input singlecell data: 
AnnData object with n_obs × n_vars = 1691 × 19972
    obs: 'cell_types'
My input ST data: 
AnnData object with n_obs × n_vars = 3067 × 135
    obsm: 'location'

However, when training the sc model, the following error occurred:

```
ValueError: Expected parameter loc (Tensor of shape (128, 5)) of distribution Normal(loc: torch.Size([128, 5]), scale: torch.Size([128, 5])) to satisfy the constraint Real(), but found invalid values:
tensor([[nan, nan, nan, nan, nan],
        nan, nan, nan, nan, nan],
        [nan, nan, nan, nan, nan]], device='cuda:0', grad_fn=<AddmmBackward0>)
```

Have you faced with similar error before? Hoping that you can help me, thank you so much!

**Comment by cane11** [comment]
Please open this in our discourse forum https://discourse.scverse.org. It most likely has to do with your input data and is related to training scvi. scvi expects unnormalized/unscaled raw counts as input and you should do filtering for min_counts. Please report with it np.max(data.X),np.min(data.X), np.all(np.isfinite(data.X)), np.min(data.X.sum(1)).

**Comment by QiaolinLu** [comment]
> Hello, thank you for the great work, I'm trying to run DestVI on other datasets, and I've transformed all the data into h5ad format with all necessary values to train the model,
> 
> I'm following [DestVI_tutorial.ipynb](https://github.com/scverse/scvi-tutorials/blob/master/DestVI_tutorial.ipynb) to run the training in order to get the cell type composition matrix
> 
> My input singlecell data: AnnData object with n_obs × n_vars = 1691 × 19972 obs: 'cell_types' My input ST data: AnnData object with n_obs × n_vars = 3067 × 135 obsm: 'location'
> 
> However, when training the sc model, the following error occurred:
> 
> ```
> ValueError: Expected parameter loc (Tensor of shape (128, 5)) of distribution Normal(loc: torch.Size([128, 5]), scale: torch.Size([128, 5])) to satisfy the constraint Real(), but found invalid values:
> tensor([[nan, nan, nan, nan, nan],
>         nan, nan, nan, nan, nan],
>         [nan, nan, nan, nan, nan]], device='cuda:0', grad_fn=<AddmmBackward0>)
> ```
> 
> Have you faced with similar error before? Hoping that you can help me, thank you so much!

yeah, it works fine in CondSCVI, but it runs with ValueError in DestVI training. I try to change the learning rate,  to  use raw data input or log-normalized data input, the bug remains. 
Have you solved the error?  I really hope that you can help me, thank you!

**Comment by izabellaleahz** [comment]
I am also having the same issue, any insight on how to fix this bug would be helpful.
